validphys2/src/validphys/covmats.py

```diff
@@ -340,6 +344,186 @@ def dataset_inputs_t0_covmat_from_systematics(
     )
 
 
+def dataset_inputs_t0_total_covmat_separate(dataset_inputs_loaded_cd_with_cuts,
+    *,
+    data_input,
+    use_weights_in_covmat=True,
+    norm_threshold=None,
+    dataset_inputs_t0_predictions,
+    loaded_theory_covmat,
+    ):
+    """
+    Function to compute the total covmat to be used for the chi2 by fitting_data_dict.
+    Depending on the theory_covmat_flag value, the theory covmat will be added to the 
+    experimental covmat or not. With the use_t0_fitting flag it is possible to choose
+    if using the t0 prescription in the covmat computation. In order to compute the covmat
+    only using the additive errors, you can set only_add to True.
+    """
+    covmat = generate_exp_covmat(dataset_inputs_loaded_cd_with_cuts, data_input, use_weights_in_covmat, norm_threshold, dataset_inputs_t0_predictions , True)
```

Comment:
Yes but I don't think it would be simpler in any case. Moreover, maybe it would become more difficult to follow. However, if you prefer I can change it.

Reply:
If you use `dataset_inputs_t0_exp_covmat_separate` then `vp` will read up the arguments and find out whether it can run that part in particular and let you know that that's failed (instead of saying that it was `total_covmat`).

In addition it could be cached so you can potentially save calls to `generate_exp_covmat`.

Reply:
Ok then I'll change